pyracf/common/security_result.py

```diff
@@ -56,6 +63,42 @@ def __extract_info(self) -> None:
             info.append(item.text)
             self.__definition.remove(item)
 
+    def __check_result_type(self, result: dict) -> dict:
+        not_profiles = ["returnCode", "reasonCode", "runningUserid"]
+        for profile_type in result["securityResult"]:
+            if profile_type in not_profiles:
+                continue
+            if "error" in result["securityResult"][profile_type]:
+                return self.__organize_smo_error(result, profile_type)
+        return result
+
+    def __organize_smo_error(self, result: dict, profile_type: str) -> dict:
```

Comment:
Can we change the name of this to `__standardize_error_response`? It could be good to include a short docstring to briefly explain why this post processing is needed.

Reply:
Also, it probably makes sense to do this earlier in `SecurityResult.__extract_error`, no? Then there doesn't need to be a special extra post processing step. All of this can more or less just be handled in `SecurityResult.__extract_error`.

Reply:
The more I think about this, the more I think that smo_error should be lumped in with null_response_error rather than security_request_error and both their names should be changed.

These error objects that have been "standardized" to look like result dictionaries SHOULD have a different layout. There are no commands that were generated or run, more information may be available in the IRRSMO00 return and reason codes, and it makes more sense for that to be a SecurityRequestError because that means there was an error with the fields or authorities to create and/or run the security request that we sent to IRRSMO00.

What we have been using as the "standard" for SecurityRequestError is REALLY IRRSMO00 executed our security request successfully, but RACF issued some warnings or errors based on the commands generated from the Request (I thought about calling this SecurityResultError).

In short, you are right that it should be handed that way if we are putting these together as one error, but I no longer think that should be the case.

Reply:
I avoided that in the past because it also would impose a lot of changes to unit testing, but I think it is necessary.

Reply:
I figured out a way to do it that didn't break much of anything so I have now committed my proposed change to the PR.

Reply:
I generally agree with this change. The only thing I am disagreement with is the naming. `SecurityRequestError` is already established as the exception that gets raised when a "security request" fails so that shouldn't change. So, when no request is able to be made we need a name that indicates that no request was able to be made. If I understand correctly, these errors occur as a result of IRROSM00's input validation. So maybe we can call this exception something like `DownstreamInputValidationError` to indicate that it was not pyRACF's input validation that failed, which we already have exceptions for, but instead it was IRRSMO00's inputs validation that happens downstream that fails. Let me know what you think.

Reply:
That then also leaves the naming for when we get a "null" response from IRRSMO00. That exception should maybe be called something like `DownstreamFatalError` to indicate that something unrelated to the user's input or the pyRACF installation is preventing the request from being processed.